app/controllers/users/two_factor_setup_controller.rb

```diff
@@ -1,65 +1,49 @@
 module Users
   class TwoFactorSetupController < TwoFactorVerificationController
     before_action :authenticate_user!
-    before_action :confirm_two_factor!
-    before_action :verify_direct_otp_sent, only: [:confirm, :update]
+    before_action :redirect_confirmed_users
 
-    # Display view to send one-time-password or input authenticator code
-    def new
+
+    def index
     end
 
 
-    # Send SMS and redirect to confirm if valid number
-    def create
-      unless params[:phone_number].nil?
-        current_user.unconfirmed_phone_number =  params[:phone_number]
+    def choice
+      session[:setup_2fa] = []
+      redirect_to users_two_factor_setup_path and return if params[:validation].nil?
```

Comment:
Good idea, added some simple JS and error handling if form submission happens anyway.